app/default/account_routes.py

```diff
@@ -19,52 +22,96 @@
+    READY_TO_SUBMIT
+
+    :param applications: List of applications to update statuses for
+    :param round_status: Round status object, used in determining the
+    display status
+    """
+    for application in applications:
+        if round_status.past_submission_deadline:
+            if application.status != "SUBMITTED":
+                application.status = "NOT_SUBMITTED"
+        else:
+            if application.status == "COMPLETED":
+                application.status = "READY_TO_SUBMIT"
+    return applications
+
+
+def build_application_data_for_display(
+    applications: list[ApplicationSummary], visible_fund_short_name
+):
     application_data_for_display = {
         "funds": [],
         "total_applications_to_display": 0,
     }
-
-    all_funds = get_all_funds()
-    if not all_funds:
-        return application_data_for_display
     count_of_applications_for_visible_rounds = 0
 
-    for fund in all_funds:
+    funds_to_show = get_visible_funds(visible_fund_short_name)
+    for fund in funds_to_show:
         fund_id = fund["id"]
         all_rounds_in_fund = get_all_rounds_for_fund(fund_id, as_dict=False)
         fund_data_for_display = {
             "fund_data": fund,
             "rounds": [],
         }
         for round in all_rounds_in_fund:
-            round_id = round.id
             round_status = determine_round_status(round)
             apps_in_this_round = [
-                app for app in applications if app.round_id == round_id
+                app for app in applications if app.round_id == round.id
             ]
             if round_status.not_yet_open or (
                 round_status.past_submission_deadline
                 and len(apps_in_this_round) == 0
             ):
                 continue
-            for application in apps_in_this_round:
-                if round_status.past_submission_deadline:
-                    if application.status != "SUBMITTED":
-                        application.status = "NOT_SUBMITTED"
-                else:
-                    if application.status == "COMPLETED":
-                        application.status = "READY_TO_SUBMIT"
+            apps_for_display = update_applications_statuses_for_display(
+                apps_in_this_round, round_status
+            )
             fund_data_for_display["rounds"].append(
                 {
                     "is_past_submission_deadline": round_status.past_submission_deadline,  # noqa
                     "is_not_yet_open": round_status.not_yet_open,
                     "round_details": round,
-                    "applications": apps_in_this_round,
+                    "applications": apps_for_display,
                 }
             )
-            count_of_applications_for_visible_rounds += len(apps_in_this_round)
+            count_of_applications_for_visible_rounds += len(apps_for_display)
+        if len(fund_data_for_display["rounds"]) == 0:
```

Comment:
In Python, empty collections such as lists, dictionaries, and sets are considered "falsy".  So you can directly use "if not fund_data_for_display["rounds"]:" without needing to call the "len()" function.

Reply:
done!

Reply:
+1 ^

```suggestion
        if not fund_data_for_display["rounds"]:
```

Reply:
sorted :D